client/document/support-user.jsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+
+function SupportUser( { supportUser = '', supportToken = '', authorized = false } ) {
+	return (
+		<html lang="en">
+			{ /* eslint-disable react/no-danger */ }
+			<head>
+				<script
+					dangerouslySetInnerHTML={ {
+						__html: `
```

Comment:
I personally don't love js codeblocks in string form.
Is there anything we can do to bring as much of this functionality into regular js as possible?

One trick i know is calling `.toString()` on functions like this:

```jsx
// define main functionality earlier in the file in plain js
function doSupportUserThings({ supportUser, supportToken, authorized }) {
 // do stuff
}

// then later initialize + call the function
<script dangerouslySetInnerHTML= {{__html:`
   ${doSupportUserThings.toString()};

    doSupportUserThings( {
       supportUser: ${supportUser}, 
       supportToken: ${supportToken}
       authorized: ${authorized}
    })
`}} />
```

Reply:
can we not directly embed real JS here? I thought it would carry over from server to client…

Reply:
@dmsnell do you mean having the JS as _"children"_ of `<style />`? [we can use javascript expressions wrapped](https://reactjs.org/docs/jsx-in-depth.html#javascript-expressions-as-children) in `{ }` but that would look even more confusing i'd say and I'm not sure it'd work as we want it to.

@samouri I tried what you suggested but I get problems while the code is then transpiled because it's not only a string anymore and e.g. `JSON.stringify` becomes `(0,_stringify2.default)` and that's not defined anywhere there (I assume we'd need to add our js bundle).

I really would like to see a way around this but I'm not sure we get sufficient results. If you have any further suggestions I'm open to try those out 👍

Reply:
although there would be a performance penalty of unknown impact, we could also write the JS in a separate file and `fs.readFileSync()` it here to keep the code itself out of a string (which means we don't suffer those disadvantages of things like no linting and no syntax highlighting or autocomplete)

Reply:
While I think this is a viable approach we should avoid using synchronous methods. Even if this route doesn’t have high frequency the method is blocking and could affect other users.

I tried a few things and got @samouri’s approach working when including the `config` and `manifest`, `build` and `vendor` JS files (transpiled `JSON.stringify` is then defined and any dependencies resolved).

What do you think?

Reply:
I'm pretty vague on my thoughts. Just exploring.

Reply:
Me too. I updated the branch, let's iterate on that.

Reply:
Can we go for the simpler uglier solution, please?

Reply:
@gwwar what do you think about https://github.com/Automattic/wp-calypso/pull/22296#discussion_r167794577? Can we go for this improved solution or should we just move to the string version as you suggested?